### PR TITLE
Totest manager refactoring

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by coolo** [comment]
@Vogtinator @DimStar77  if that's ok with you so far, I would move the configs into a project attribute and remove config from the code.

File: ttm/manager.py [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+        root = ET.parse(f).getroot()
+        for binary in root.findall('binary'):
+            binary = binary.get('filename', '')
+            result = re.match(r'.*-([^-]*)-[^-]*.src.rpm', binary)
+            if result:
+                return result.group(1)
+
+        raise NotFoundException("can't find %s version" % self.project)
+
+    def current_sources(self):
```
Comment (Vogtinator) [inline ttm/manager.py]: I don't like the "source" naming here - it should at least contain the word "version" in one way or another.

File: ttm/manager.py [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
+        if self.project.take_source_from_product is None:
+            raise Exception('No idea where to take the source version from')
+
+        if self.project.take_source_from_product:
+            if self.project.is_image_product:
```
Comment (Vogtinator) [inline ttm/manager.py]: This could be avoided by just checking whether `main_products` is empty.

File: ttm/manager.py (hunk 1 above)
Comment (coolo) [reply ttm/manager.py]: I found there were too many versions - so I wanted something different than 'current_version'. Meanwhile I'd go with version_from_project

File: ttm/manager.py (hunk 2 above)
Comment (coolo) [reply ttm/manager.py]: I can do that.

**Comment by coolo** [comment]
The dry runs were all fine, so I'd move forward.

File: ttm/manager.py [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+        raise NotFoundException("can't find %s version" % self.project)
+
+    def version_from_project(self):
+        if not self.project.take_source_from_product:
+            return self.release_version()
+
+        if len(self.project.main_products):
+            return self.iso_build_version(self.project.name, self.project.main_products[0])
+
+        #assert(self.project.is_image_product)
```
Comment (Vogtinator) [inline ttm/manager.py]: Should be removed

Comment (coolo) [reply ttm/manager.py]: it's not a leftover debug - it's a comment

Comment (Vogtinator) [reply ttm/manager.py]: It's a commented out leftover debug statement then - this assert can't possibly be useful in production.

Comment (coolo) [reply ttm/manager.py]: Right, like most comments only help the reader to understand the logic - not the production

Comment (coolo) [reply ttm/manager.py]: You might be confused by this comment being pseudo code, but that's only because it's easier to express this way

Comment (coolo) [reply ttm/manager.py]: I reworked that whole image_product thingy now

**Comment by Vogtinator** [comment]
Looking good - I just don't like this being implicit:

```
        if not len(self.project.main_products):
            version_file = version_file + '_images'
```

Simply by not having an empty `main_products` array it picks an entirely different version file. I don't have a better idea though.

**Comment by Vogtinator** [comment]
Actually, it might be possible to delete that now - the `:Images` subprjs have their own `00Meta/version_*` now.

**Comment by coolo** [comment]
let's keep more refactorings for another round. This was just a precondition for the real split

**Pull request merged**